desktop/ui/src/main/java/org/datacleaner/widgets/result/CategorizationResultAbstractSwingRenderer.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.widgets.result;
+
+import java.awt.FlowLayout;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.swing.Box;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.table.DefaultTableModel;
+
+import org.datacleaner.api.AnalyzerResult;
+import org.datacleaner.api.Provided;
+import org.datacleaner.bootstrap.WindowContext;
+import org.datacleaner.panels.DCPanel;
+import org.datacleaner.result.AnnotatedRowsResult;
+import org.datacleaner.result.CategorizationResult;
+import org.datacleaner.result.renderer.AbstractRenderer;
+import org.datacleaner.result.renderer.RendererFactory;
+import org.datacleaner.util.IconUtils;
+import org.datacleaner.util.LabelUtils;
+import org.datacleaner.util.WidgetFactory;
+import org.datacleaner.windows.DetailsResultWindow;
+import org.jfree.data.general.DefaultPieDataset;
+
+public abstract class CategorizationResultAbstractSwingRenderer<R extends CategorizationResult> extends AbstractRenderer<R, JComponent> {
```

Comment:
Naming-wise I much prefer to put Abstract as first, so it becomes AbstractCategorizationResultSwingRenderer